sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -270,14 +410,45 @@ Context getContext() {
      * @throws NullPointerException if {@code key} or {@code value} is null.
      * @return The updated {@link EventData}.
      */
-    public EventData addContext(String key, Object value) {
+    EventData addContext(String key, Object value) {
         Objects.requireNonNull(key, "The 'key' parameter cannot be null.");
         Objects.requireNonNull(value, "The 'value' parameter cannot be null.");
         this.context = context.addData(key, value);
 
         return this;
     }
 
+    /**
+     * Checks the length of ID fields.
+     *
+     * Some fields within the message will cause a failure in the service without enough context information.
+     */
+    private void checkIdLength(String fieldName, String value, int maxLength) {
+        if (value != null && value.length() > maxLength) {
+            final String message = String.format("%s cannot be longer than %d characters.", fieldName, maxLength);
+            throw logger.logExceptionAsError(new IllegalArgumentException(message));
+        }
+    }
+
+    /**
+     * Validates that the user can't set the partitionKey to a different value than the session ID. (this will
+     * eventually migrate to a service-side check)
+     */
+    private void checkPartitionKey(String proposedPartitionKey) {
```

Comment:
This is not correct for Event Hubs;  this logic is specific to Service Bus.   We should be treating the partition key as an opaque bit of data.

Reply:
If we treat the partition key as an opaque data bit, should the partition key data and the Getter method also be considered invisible?

Reply:
According to your instructions, fix in new version.